### DISCUSSION TITLE
Branch Name Wildcard Support

### DISCUSSION THREAD

**Discussion by hdalaq** [BetaHuhn/repo-file-sync-action#83] [Ideas]
I've ran into few cases where I wished wildcards were possible in the branch names! And was wondering if anyone sees a value in supporting it! Something like the following.

```
group:
  repos: |
    foo/bar@*/release/*
    foo/bar@*/docs/*
    foo/bar@*/pages/*
```

**Comment by BetaHuhn** [comment]
Just to understand it better, with `foo/bar@*/release/*` we would sync the files/directories to all the branches of the `foo/bar` repo where the name begins with something then contains `/release/` and then ends with something?

**Reply by hdalaq** [reply]
I guess a better example is syncing a workflow for all feature branches. So instead of doing `foo/bar@feature/a` and `foo/bar@feature/b` and so on we can use something like foo/bar@features/*` that will run it on all the feature branches.

**Reply by BetaHuhn** [reply]
Got it! I can definitely see the use for a feature like this.

To get this working we would have to get all branches for the repo using GitHub's API and then filter through them based on the selector. This could be done in the [config parsing phase](https://github.com/BetaHuhn/repo-file-sync-action/blob/f3a75bbcb97a8e120b1520e49d749f8750f14a02/src/config.js#L163).

Do you feel comfortable creating a PR for this @hdalaq?
